strings-csharp/GetFirstNCharactersOfAString/GetFirstNCharactersOfAString/FirstNCharactersOfStringGetter.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Text;
+
+namespace GetFirstNCharactersOfAString;
+
+public class FirstNCharactersOfStringGetter(string inputString = "CodeMaze", int numberOfCharacters = 4)
+{
+    private bool IsInputStringShorter()
+        => inputString.Length < numberOfCharacters;
+
+    public ReadOnlySpan<char> UseForLoop()
+    {
+        if (IsInputStringShorter())
+            return inputString;
+
+        var strBuilder = new StringBuilder(numberOfCharacters);
+        for (int i = 0; i < numberOfCharacters; i++)
+        {
+            strBuilder.Append(inputString[i]);
+        }
+
+        return strBuilder.ToString();
+    }
+
+    public ReadOnlySpan<char> UseRemove()
+    {
+        if (IsInputStringShorter())
+            return inputString;
+
+        return inputString.Remove(numberOfCharacters);
+    }
+
+    public ReadOnlySpan<char> UseLINQ()
+        => string.Join("", inputString.Take(numberOfCharacters));
```

Comment:
@cjaynjoku 
This method takes the N character and converts it to `string` before returning as `ReadOnlySpan<>`.
I think we can get better performance by just returning the `IEnumerable` since we are focusing on returning number of characters.
I made the change and ran the benchmark and got:

![Screenshot_1](https://github.com/CodeMazeBlog/CodeMazeGuides/assets/13255247/5e040526-fcb9-4877-9a4e-76a83cb79ce3)
The new method is the `UseLINQEnumerable` method. Though it uses more memory than the next 2 methods
@jshergal what do you think?

Reply:
@elfico - I agree with you here. Since the article is focused on getting the first N characters, we don't have to return them as strings, or spans. It could be an array, it could be an IEnumerable, etc. The key is that we return "N" characters.

Note, that regarding an IEnumerable, in the benchmark we need to be sure to actually consume it:
https://mawosoft.github.io/BenchmarkDotNet/api/BenchmarkDotNet.Engines.ConsumerExtensions.html

Reply:
@cjaynjoku 
Like Jeff mentioned please make a note to use `Consume`. When I added the `Consume` construct, the `UseLINQEnumerable` performed worse.
Here is a link to the documentation: https://benchmarkdotnet.org/articles/samples/IntroDeferredExecution.html

Reply:
I have already done this @elfico. Please, check the latest commit. Thank you.